AirCasting/SessionViews/SessionHeaderView.swift

```diff
@@ -72,24 +72,35 @@ private extension SessionHeaderView {
             //  |_________|     |-------------------|
             // so the idea at leat for now is this below
             #warning("Fix - Handle session.deviceType (for now it is always nill)")
-            switch session.type?.description {
+            sensorType
+                .font(Font.moderate(size: 13, weight: .regular))
+        }
+        .foregroundColor(.darkBlue)
+    }
+    
+    var sensorType: some View {
+        switch session.type.description {
             case "Fixed":
-                Text("\(session.type!.description) : AirBeam3")
-                    .font(Font.moderate(size: 13, weight: .regular))
+                return airBeamText
             case "Mobile":
-                if session.allStreams!.count > 1 {
-                    Text("\(session.type!.description): AirBeam3")
-                        .font(Font.moderate(size: 13, weight: .regular))
+                if session.allStreams!.count == 1 {
+                    return Text("\(session.type!.description): \(Strings.SessionHeaderView.mic)")
                 } else {
-                    Text("\(session.type!.description): Phone Mic")
-                        .font(Font.moderate(size: 13, weight: .regular))
+                    return airBeamText
                 }
             default:
-                Text(session.deviceType?.description ?? "")
-                    .font(Font.moderate(size: 13, weight: .regular))
-            }
+                return Text(session.deviceType?.description ?? "")
+        }
+    }
+    
+    var airBeamText: Text {
+        if session.pm1Stream?.sensorName?.contains(Strings.SessionHeaderView.airBeam1) != nil {
```

Comment:
this is really convoluted and a bit weird (like - why are you using pm1Stream here? -> it's getting a bit clear after really focusing on reading this piece of code, but the fact that this question is asked is already a big tell that something is maybe under-designed).

Another pitfall - imagine there is an AB4 introduce like a year from now. Will you remember to go here and add another `else if`? This should really be more universal than this or at least be future proof in some way.